eth-connector/src/admin_controlled.rs

```diff
@@ -49,10 +49,7 @@ pub trait AdminControlled {
     ///
     /// Error is returned if the caller hasn't rights.
     fn assert_access_right(&self) -> Result<(), error::AdminControlledError> {
-        if self.get_access_right() == near_sdk::env::predecessor_account_id()
-            || self.is_owner()
-            || near_sdk::env::predecessor_account_id() == near_sdk::env::current_account_id()
-        {
+        if self.get_access_right() == near_sdk::env::predecessor_account_id() || self.is_owner() {
```

Comment:
The name `get_access_right` is not correct, imo. I would rename it to `get_account_with_access_right`. Or create method `has_access_right(account_id: &Account) -> bool` instead.